### ISSUE TITLE
Android icon not taken from config

### ISSUE THREAD

**Issue by MatanYed** [phonegap/phonegap-plugin-push#2598]
### Expected Behaviour
To use the icon that specified in the config in init()
### Actual Behaviour
using the default icon of the app
#### Steps to Reproduce
config.xml:

    <platform name="android">
        <resource-file src="resources/android/push/drawable-ldpi-icon.png" target="app/src/main/res/drawable-ldpi/push_icon.png" />
        <resource-file src="resources/android/push/drawable-mdpi-icon.png" target="app/src/main/res/drawable-mdpi/push_icon.png" />
        <resource-file src="resources/android/push/drawable-hdpi-icon.png" target="app/src/main/res/drawable-hdpi/push_icon.png" />
        <resource-file src="resources/android/push/drawable-xhdpi-icon.png" target="app/src/main/res/drawable-xhdpi/push_icon.png" />
        <resource-file src="resources/android/push/drawable-xxhdpi-icon.png" target="app/src/main/res/drawable-xxhdpi/push_icon.png" />
        <resource-file src="resources/android/push/drawable-xxxhdpi-icon.png" target="app/src/main/res/drawable-xxxhdpi/push_icon.png" />
        <icon density="ldpi" src="resources/android/icon/drawable-ldpi-icon.png" />
        <icon density="mdpi" src="resources/android/icon/drawable-mdpi-icon.png" />
        <icon density="hdpi" src="resources/android/icon/drawable-hdpi-icon.png" />
        <icon density="xhdpi" src="resources/android/icon/drawable-xhdpi-icon.png" />
        <icon density="xxhdpi" src="resources/android/icon/drawable-xxhdpi-icon.png" />
        <icon density="xxxhdpi" src="resources/android/icon/drawable-xxxhdpi-icon.png" />
    </platform>

app.js:

	const push = window.PushNotification.init({
		android: {
			icon: 'push_icon',
			iconColor: '#39607A'
		},
		ios: {
			alert: "true",
			badge: "true",
			sound: "true"
		}
	});

the default icon is square non-transparent icon, the push_icon is round with transparent parts, but the push notifications still are square gray, which means it using the default icon and not the one i specified in the app.js
#### Platform and Version (eg. Android 5.0 or iOS 9.2.1)
Android 8.1
#### (Android) What device vendor (e.g. Samsung, HTC, Sony...)
Xiaomi
#### Cordova CLI version and cordova platform version
cordova 8.1.2
cordova-android 7.1.1

#### Plugin version
phonegap-plugin-push 2.2.3 "PushPlugin"

**Comment by andreszs** [comment]
Where exactly is the PNG icon file located? Please share the code you are using to copy this icon during build time, either by a hook or config file setting. [Here are samples](https://stackoverflow.com/questions/35511791/adding-drawable-resource-using-cordova) of using both these methods. If your icon is not copied properly to the right path, it wont show up.

**Comment by MatanYed** [comment]
Relative path: `/platforms/android/app/src/main/res/drawable-hdpi/push_icon.png`
The files copied successfully.

**Comment by andreszs** [comment]
Assuming you are using **FCM** and not GCM, this is what I'm doing now and it works:

- Remove `icon` and `iconColor` from the app.js file.
- Add `icon` and `color` to your push payload, along with your `title`, `body`, and `sound` parameters.

The notification will show your icon as long as it's located in the path you have shown. My icons are located in: "platforms/android/res/drawable-port-mdpi/push.png" and other DPI paths.

This is my FCM payload base in PHP:
```
$notification = array();
$notification['title'] = (isset($data['title']) ? $data['title'] : null);
$notification['body'] = (isset($data['message']) ? $data['message'] : null);
$notification['sound'] = 'default';
$notification['tag'] = 1; // reusar mismo tag para reemplazar notificacion anterior
$notification['icon'] = 'push'; // means push.png icon in drawable
$notification['color'] = '#00A3D6'; // icon bg color for Android 5+
```

Notice: I'm using plugin 2.1.3 and cordova-android 6.3.0.

**Comment by MatanYed** [comment]
It's kinda problematic solution, but thank you for helping. I guess i'll use it if there is not any other solution...

**Comment by andreszs** [comment]
Yes, but this has been defined in the [FCM HTTP v1 API](https://firebase.google.com/docs/reference/fcm/rest/v1/projects.messages#androidnotification), as you can see from the FCM App Server Protocols documentation linked.

Most of the plugin docs here in GitHub refer to the obsolete GCM protocol, and no one is updating it to match the FCM standard. This method looks ackward, but it's actually FCM's way.

![image](https://user-images.githubusercontent.com/10298071/47616298-93d71d00-da99-11e8-8a36-e28fd11b4024.png)

**Comment by andreszs** [comment]
The app.js method was for the obsolete, deprecated GCM method -- not FCM. Setting the icon in the push itself is actually a better idea, because you can set a different icon and color, depending on the notification event sent.

**Comment by macdonst** [comment]
@MatanYed can you get a log of an incoming push using `adb logcat | grep Push`. I just want icon it is trying to load.

**Comment by Edmu** [comment]
Hello, I have run into some issue. The workaround that worked for me:

<platform name="android">
    <resource-file src="resources/android/notifs/mdpi.png" target="app/src/main/res/drawable-mdpi/ic_stat_notify.png"/>
    ...
    <resource-file src="resources/android/notifs/xxxhdpi.png" target="app/src/main/res/drawable-xxxhdpi/ic_stat_notify.png"/>
    <config-file parent="/manifest/application" target="AndroidManifest.xml">
      <meta-data android:name="com.google.firebase.messaging.default_notification_icon" android:resource="@drawable/ic_stat_notify"/>
    </config-file>
    ...
</platform>

cordova-lib@8.1.1
cordova-android 7.1.4
phonegap-plugin-push 2.2.3 "PushPlugin"

for lower cordova-android versions (6.x.x) resource-file and config-file attribute paths should be different.

**Comment by luidiva** [comment]
> Hello, I have run into some issue. The workaround that worked for me:
> 
> `<platform name="android"> <resource-file src="resources/android/notifs/mdpi.png" target="app/src/main/res/drawable-mdpi/ic_stat_notify.png"/> ... <resource-file src="resources/android/notifs/xxxhdpi.png" target="app/src/main/res/drawable-xxxhdpi/ic_stat_notify.png"/> <config-file parent="/manifest/application" target="AndroidManifest.xml"> <meta-data android:name="com.google.firebase.messaging.default_notification_icon" android:resource="@drawable/ic_stat_notify"/> </config-file> ... </platform>`
> 
> cordova-lib@8.1.1
> cordova-android 7.1.4
> phonegap-plugin-push 2.2.3 "PushPlugin"
> 
> for lower cordova-android versions (6.x.x) resource-file and config-file attribute paths should be different.

I have same configurations as yours.

When I choose targetSdkVersion > 20 icon stops showing even with your config. The files are generated in target folders, but still is not showing.
Any ideas?

Thank you

**Comment by Edmu** [comment]
after the build AndroidManifest.xml (android/app/src/main) should contain this tag:

<meta-data android:name="com.google.firebase.messaging.default_notification_icon" android:resource="@drawable/ic_stat_notify" />

this is added by adding 

<meta-data android:name="com.google.firebase.messaging.default_notification_icon" android:resource="@drawable/ic_stat_notify"/> </config-file>

into config.xml

also notification icon files after the build should be in place:

android/app/src/main/res/drawable.../icon_name.png

this is done by adding tags like:

<resource-file src="resources/android/notifs/mdpi.png" target="app/src/main/res/drawable-mdpi/ic_stat_notify.png"/>

into config.xml it tells to copy over file from src to target dir
